File: behaviorial/chain-of-responsability/python/concrete_handler.py
from typing import Optional
from base_handler import BaseHandler


class ConcreteHandler1(BaseHandler):
    def handle(self, request: str) -> Optional[str]:
        if request == 'Request1':
            return f'ConcreteHandler1 handled the request: {request}'
        return super().handle(request)


class ConcreteHandler2(BaseHandler):
    def handle(self, request: str) -> Optional[str]:
        if request == 'Request2':
            return f'ConcreteHandler2 handled the request: {request}'
        return super().handle(request)


class ConcreteHandler3(BaseHandler):
    def handle(self, request: str) -> Optional[str]:
        if request == 'Request3':
            return f'ConcreteHandler3 handled the request: {request}'
        return super().handle(request)

File: behaviorial/chain-of-responsability/typescript/concrete-handler.ts
import { BaseHandler } from './base-handler';

export class ConcreteHandler1 extends BaseHandler {
	public handle(request: string): string | null {
		if (request === 'Request1') {
			return `ConcreteHandler1 handled the request: ${request}`;
		}
		return super.handle(request);
	}
}

export class ConcreteHandler2 extends BaseHandler {
	public handle(request: string): string | null {
		if (request === 'Request2') {
			return `ConcreteHandler2 handled the request: ${request}`;
		}
		return super.handle(request);
	}
}

export class ConcreteHandler3 extends BaseHandler {
	public handle(request: string): string | null {
		if (request === 'Request3') {
			return `ConcreteHandler3 handled the request: ${request}`;
		}
		return super.handle(request);
	}
}
